guides/search.md

```diff
@@ -0,0 +1,381 @@
+    .auth(credentials)
+    .build()?;
+let client = OpenSearch::new(transport);
+client
+  .indices()
+  .create(IndicesCreateParts::Index("movies"))
+  .send()
+  .await?;
+
+for n in 0..11 {
+    client
+        .index(IndexParts::IndexId("movies", &i.to_string()))
+        .body(json!({
+            "title": format!("The Dark Knight {}", i),
+            "director": "Christopher Nolan",
+            "year": 2008 + i
+        }))
+        .send()
+        .await?;
+}
+
+client
+    .index(IndexParts::IndexId("movies", &i.to_string()))
+    .body(json!({
+        "title": "The Godfather",
+        "director": "Francis Ford Coppola",
+        "year": "1972"
+    }))
+    .send()
+    .await?;
+
+client
+    .index(IndexParts::IndexId("movies", &i.to_string()))
+    .body(json!({
+        "title": "The Shawshank Redemption",
+        "director": "Frank Darabont",
+        "year": "1994"
+    }))
+    .send()
+    .await?;
+
+// Refresh the index to make the documents searchable
+client
+    .indices
+    .refresh(IndicesRefreshParts::Index(&["movies"]))
+    .send()
+    .await?
+```
+
+## Search API
+
+### Basic Search
+
+The search API allows you to search for documents in an index. The following example searches for ALL documents in the `movies` index:
+
+```rust
+let response = client
+    .search(SearchParts::Index(&["movies"]))
+    .send()
+    .await?
+    .json::<Value>()
+    .await?;
+println!("{}", response["hits"]["count"]);
+```
+
+You can also search for documents that match a specific query. The following example searches for documents that match the query `dark knight`:
+
+```rust
+let response = client
+    .search(SearchParts::Index(&["movies"]))
+    .body(json!({
+        "query": {
+            "multi_match": {
+                "query": "dark knight",
+                "fields": ["title"]
+            }
+        }
+    }))
+    .send()
+    .await?
+    .json::<Value>()
+    .await?;
+println!("{:?}", response["hits"]["hits"].as_array().unwrap());
+```
+
+OpenSearch query DSL allows you to specify complex queries. Check out the [OpenSearch query DSL documentation](https://opensearch.org/docs/latest/query-dsl/) for more information.
+
+### Basic Pagination
+
+The search API allows you to paginate through the search results. The following example searches for documents that match the query `dark knight`, sorted by `year` in ascending order, and returns the first 2 results after skipping the first 5 results:
+
+```rust
+let search_body = json!({
+    "query": {
+        "match": {
+            "title": "dark knight"
+        }
+    },
+    "sort": [
+        {
+            "year": {
+                "order": "asc"
+            }
+        }
+    ]
+});
+
+let response = client
+    .search(SearchParts::Index(&["movies"]))
+    .from(5)
+    .size(2)
+    .body(search_body)
+    .send()
+    .await?
+    .json::<Value>()
+    .await?;
+println!("{:?}", response["hits"]["hits"].as_array().unwrap());
```

Comment:
There's no reason to keep `search_body` as a separate variable, so just inline it:
```suggestion
let response = client
    .search(SearchParts::Index(&["movies"]))
    .from(5)
    .size(2)
    .body(json!({
        "query": {
            "match": {
                "title": "dark knight"
            }
        },
        "sort": [
            {
                "year": {
                    "order": "asc"
                }
            }
        ]
    }))
    .send()
    .await?
    .json::<Value>()
    .await?;
println!("{:?}", response["hits"]["hits"].as_array().unwrap());
```

Reply:
@Xtansia We are using `search_body` in 2 places, why do we need to duplicate the same body data, instead of storing in a single variable and reusing it?

Reply:
@sayuree Because the value is "moved" out of the variable by `.body(search_body)`, meaning the second time you reference it it's no longer valid. This is clear if you try to compile the code in an example file as I'd suggested originally. The only way it works to re-use it is to use `search_body.clone()`, but I feel that  is less clear due to how disconnected the final usage is from the first.